internal/response-consumer/handler_test.go

```diff
@@ -240,83 +273,143 @@ var _ = Describe("handler", func() {
 			Expect(hosts[0].Log).To(Equal("a1b2"))
 			Expect(hosts[1].Log).To(Equal("a1b2"))
 		})
+	})
 
+	Describe("correlation", func() {
+		It("updates the correct run", func() {
+			data := []dbModel.Run{
+				test.NewRun(accountNumber()),
+				test.NewRun(accountNumber()),
+			}
+
+			Expect(db().Create(&data).Error).ToNot(HaveOccurred())
+
+			events := createRunnerEvents(
+				messageModel.EventExecutorOnStart,
+				"playbook_on_start",
+				"playbook_on_play_start",
+				"playbook_on_task_start",
+				"runner_on_start",
+				"runner_on_ok",
+				"playbook_on_stats",
+			)
+
+			msg := newRunnerResponseMessage(events, data[1].CorrelationID)
+			instance.onMessage(test.TestContext(), msg)
+
+			run0 := fetchRun(data[0].ID)
+			Expect(run0.Status).To(Equal("running"))
+			hosts := fetchHosts(data[0].ID)
+			Expect(hosts).To(BeEmpty())
+
+			run1 := fetchRun(data[1].ID)
+			Expect(run1.Status).To(Equal("success"))
+			checkHost(data[1].ID, "success", nil, "", nil)
+		})
+	})
+
+	Describe("Satellite", func() {
 		It("updates the run status based on successful satellite events", func() {
 			var data = test.NewRun(accountNumber())
 			Expect(db().Create(&data).Error).ToNot(HaveOccurred())
 
-			events := createSatEvents(
+			inventoryId := uuid.New()
+			var hostData = test.NewRunHost(data.ID, "running", &inventoryId)
+			inventoryIdString := inventoryId.String()
```

Comment:
Nice. Do not forget to also change the example payloads, etc.

Reply:
Added with the validator PR :+1: